File: problems_026to050/problem_027/solution.py
import os
import sys

PROJECT_PATH = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
sys.path.append(PROJECT_PATH)

from measurement import Stopwatch
from toolbox import is_prime, smaller_primes


class Solution(Stopwatch):
    def __init__(self):
        super().__init__()

    def execute(self):
        answer = None
        max_prime_count = 0
        for b in smaller_primes(1001):
            for a in range(-b+1, 1000):
                n = 0
                while is_prime(n**2 + a*n + b):
                    n += 1
                if max_prime_count < n:
                    max_prime_count = n
                    answer = a * b
        return answer

if __name__ == '__main__':
    soln = Solution()
    result = soln.execute()
    print(result)

File: problems_026to050/problem_027/solution.ts
import { isPrime } from 'mathjs';

import { SolutionFunction } from '../../lib/types';

const solution: SolutionFunction = () => {
  let answer = null;

  let maxPrimeCount = 0;
  let b = 2;

  while (b < 1001) {
    if (isPrime(b)) {
      for (let a = -b + 1; a < 1000; a += 1) {
        let n = 0;

        while (isPrime(n ** 2 + a * n + b)) {
          n += 1;
        }

        if (maxPrimeCount < n) {
          maxPrimeCount = n;
          answer = a * b;
        }
      }
    }

    if (b === 2) {
      b += 1;
    }
    else {
      b += 2;
    }
  }

  return answer;
};

export default solution;
